Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWindowsCocoa.mm

```diff
@@ -562,6 +563,50 @@
     return *m_onFocusChanged;
 }
 
+inline WebExtensionAPIWindows::WindowTypeFilter toWindowTypeFilter(WebExtensionWindow::Type type)
+{
+    switch (type) {
+    case WebExtensionWindow::Type::Normal:
+        return WebExtensionAPIWindows::WindowTypeFilter::Normal;
+
+    case WebExtensionWindow::Type::Popup:
+        return WebExtensionAPIWindows::WindowTypeFilter::Popup;
+    }
+}
+
+void WebExtensionContextProxy::dispatchWindowsEvent(WebExtensionEventListenerType type, std::optional<WebExtensionWindowParameters> windowParameters)
```

Comment:
is it more preferable to define these dispatch methods in the .cocoa file of the API instead of in WebExtensionContextProxy?

Reply:
They could be in the cpp file, but it is often easier to put in the Cocoa file to be with the other code. At some point we will want to migrate things to the cpp files that don't use ObjC.

Reply:
ah, i see you're cleaning up the code to move all these dispatch events into the .cocoa files [here](https://github.com/WebKit/WebKit/pull/17331/commits/b681afb24fade6c6f63d8ce1a1c7c14d359d2cae)